.github/workflows/actions.yml

```diff
@@ -261,7 +261,7 @@ jobs:
             -G Ninja \
             -DCMAKE_BUILD_TYPE=RelWithDebInfo \
             -DCMAKE_INSTALL_PREFIX=pfx \
-            -DCMAKE_OSX_ARCHITECTURES='x86_64' \
+            -DCMAKE_OSX_ARCHITECTURES='arm64' \
```

Comment:
Why?

Reply:
I kind of explained it indirectly in the PR description: when `USE_SYSTEM_*` is ON, the dependencies will be on the architecture of the host. Upgrading from macos-12 to macos-14 changes the host from Intel to Apple Silicon, so the only architecture supported by the system dependencies becomes arm64.